### DISCUSSION TITLE
Pipewire Audio Output does not switch sample rates - always 48 kHz

### DISCUSSION THREAD

**Discussion by Tunoac** [audacious-media-player/audacious#96] [Q&A] [answered]
I'm using archlinux with pipewire. My DAC supports several sample rates, to get best quality and use them I have this configured:

```
# ~/.config/pipewire/pipewire.conf
default.clock.allowed-rates = [ 44100 48000 96000 88200 192000 ]
```
Issue with audacious:
"Pipewire Output" music is resampled always to 48000.
Switching to "PulseAudio" no resampling is done.
Switching to "Alsa Output" no resampling is done.

Expected behavior:
Pipewire output performs no resampling.

Reproduce:
have some flac files with 44,1 KHz, 48 kHz, 96 KHz (e.g. sine tone generator by Audacity),
run pw-top,
and let Audacious play the files.

Does anyone have an idea how to solve this ?

[Pipewire.txt](https://github.com/audacious-media-player/audacious/files/13664319/Pipewire.txt)
[PulseAudio.txt](https://github.com/audacious-media-player/audacious/files/13664321/PulseAudio.txt)

**Comment by jlindgren90** [comment]
It would be a PipeWire issue, so you should ask for support from that project. Audacious does not perform any resampling automatically.

Or you could just make use of PipeWire's ALSA or PulseAudio emulation since that seems to give better results. From what I have heard, PipeWire developers expect for most applications continue to use those APIs.

**Comment by radioactiveman** [comment, marked as answer]
I could reproduce this with Audacious and Qmmp, but not mpv which sets `PW_KEY_NODE_RATE` like SDL to the actual sample rate.

Should be fixed now with https://github.com/audacious-media-player/audacious-plugins/commit/932e1bcb23957a283d4d6c25643cee91f54567fd, thanks for reporting. :+1:

**Comment by Tunoac** [comment]
Thanks very much!
I was lucky and found an archlinux aur from Radioactiveman :-)  and could fetch+compile from latest git and test it successfully.
